code/modules/projectiles/ammunition/magazines.dm

```diff
@@ -199,16 +199,16 @@
 	desc = "A gun magazine. Seems to be broken and can only hold one bullet. Pretty useless."
 	max_ammo = 1
 
+/obj/item/ammo_box/magazine/m10mm/empty/update_icon()
+	icon_state = "[initial(icon_state)]-[stored_ammo.len ? "8" : "0"]"
```

Comment:
One thing I tried was changing the code that gives the image for all 10mm with:

/obj/item/ammo_box/magazine/m10mm/update_icon()
	..()
	icon_state = "[initial(icon_state)]-[Ceiling(ammo_count(0)/8)*8]"

Its longer but after some testing, works with both the single bullet crap mag and the proper 8_bullet _mags.__